moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -165,7 +165,7 @@ void RobotState::copyFrom(const RobotState& other)
             (robot_model_->getVariableCount() * (1 + ((has_velocity_ || has_acceleration_ || has_effort_) ? 1 : 0) +
                                                  ((has_acceleration_ || has_effort_) ? 1 : 0)) +
              nr_doubles_for_dirty_joint_transforms);
-    memcpy(variable_joint_transforms_, other.variable_joint_transforms_, bytes);
+    memcpy((char*)variable_joint_transforms_, (char*)other.variable_joint_transforms_, bytes);
```

Comment:
Use a cpp style cast.

Reply:
See https://github.com/ros-planning/moveit/pull/2180#discussion_r448487601. I'd like to keep the diff minimal or we should change it for all versions. I should update it to void, though.